posthog/api/capture.py

```diff
@@ -536,7 +536,11 @@ def capture_internal(event, distinct_id, ip, site_url, now, sent_at, event_uuid=
 
     candidate_partition_key = f"{token}:{distinct_id}"
 
-    if distinct_id.lower() not in LIKELY_ANONYMOUS_IDS and is_randomly_partitioned(candidate_partition_key) is False:
+    if (
```

Comment:
```suggestion
    # If historical data we always want to set the partition key
    # If it looks legit and hasn't been too much also set the partition key
    # otherwise partition_key = None (throw it to overflow)
    if (
```